### PR TITLE
fix: Fix insufficient balance alert when other network is selected by network picker

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR aims to fix insuficient balance alert when network picker is selected other than the selected network.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/40962?quickstart=1)

## **Changelog**

CHANGELOG entry:

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/CONF-1046

## **Manual testing steps**

1. Have no gas on Polygon
2. Have gas on Ethereum
3. Set the network in the home page to Ethereum
4. Start a send on Polygon
5. Notice how you’ll be allowed to submit the transaction with no insufficient funds warning and without using gas station
6. Submitting the transaction will make it fail with insufficient funds for gas
7. Now, do the opposite
8. Set the network in the home page to Polygon
9. Start a send on Ethereum
10. Notice how gas will default to use gas station even though you have plenty of ETH for gas

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Changes the insufficient-balance check to use `useFeeCalculations`/transaction-specific max fee instead of a selector keyed by transaction id, which can affect blocking alerts and send flow across networks. Low code churn but touches fee/balance validation logic used during confirmations.
> 
> **Overview**
> Fixes insufficient-balance detection during confirmations when the user has a different network selected in the network picker by switching `useHasInsufficientBalance` to compute gas cost from `useFeeCalculations` (`maxFeeHex`) rather than `selectTransactionFeeById`.
> 
> Updates `useFeeCalculations` to expose `maxFeeHex` (and adjusts snapshots), and updates/strengthens related tests to use checksum addresses and a clearer batch-balance edge case. Jest console baselines are refreshed to match new warnings/output.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 235a4e9c216addf3ce321bd17591cc1b6cb82a28. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/pages/confirmations/hooks/useHasInsufficientBalance.ts [hunk 1]
```diff
@@ -6,10 +6,10 @@ import { sumHexes } from '../../../../shared/lib/conversion.utils';
 import {
   getMultichainNetworkConfigurationsByChainId,
   getNativeTokenCachedBalanceByChainIdSelector,
-  selectTransactionFeeById,
```
Comment (OGPoyraz) [inline ui/pages/confirmations/hooks/useHasInsufficientBalance.ts:9]: We will remove this selector and downstream selector on separate PR. We will be able to remove `ui/pages/confirmations/components/gas-details-item/gas-details-item.js` and `ui/components/app/cancel-speedup-popover/cancel-speedup-popover.js`. Syncing with @vinistevam  on this.

Comment (OGPoyraz) [reply ui/pages/confirmations/hooks/useHasInsufficientBalance.ts:9]: https://github.com/MetaMask/metamask-extension/pull/40911 will remove it - ignore this comment.

**Review by jpuri — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]
